addons/project/models/project.py

```diff
@@ -741,6 +742,8 @@ def action_view_tasks(self):
             .env.ref('project.act_project_project_2_project_task_all') \
             .sudo().read()[0]
         action['display_name'] = self.name
+        context = dict(safe_eval(action.get('context', {}), {'active_id': self.id}), create=self.active)
```

Comment:
Couldn't we rather use a `literal_eval` instead of a `safe_eval`?
Also, I know it is not from your diff, but could you replace the action getter by `self.env['ir.actions.act_window']._for_xml_id('project.act_project_project_2_project_task_all')` this is a better way of doing the same. I guess that with_context(active_id=...) is not providing the expected results and can be removed.

Thanks 🙂 !